lib/cisco_node_utils/bgp.rb

```diff
@@ -494,6 +507,35 @@ def default_reconnect_interval
       config_get_default('bgp', 'reconnect_interval')
     end
 
+    # Route Distinguisher (Getter/Setter/Default)
+    # Configure in vrf context
+    def route_distinguisher
+      return false unless RouterBgp.feature_nv_overlay_evpn_enabled
+      config_get('bgp', 'route_distinguisher', @get_args)
+    end
+
+    def route_distinguisher=(rd)
+      # In order to remove a bgp route-distinguisher you cannot simply issue
+      # 'no rd'. Dummy-id specified to work around this.
+      RouterBgp.feature_nv_overlay_evpn_enable if
+        rd != default_route_distinguisher &&
+        !RouterBgp.feature_nv_overlay_evpn_enabled
+      dummy_rd = '1:1'
+      if rd == default_route_distinguisher
+        @set_args[:state] = 'no'
+        @set_args[:rd] = dummy_rd
```

Comment:
If you actually need to specify rd (see above comment) then don't use a dummy_rd, use the getter method:

```
   @set_args[:rd] = route_distinguisher
```